h5py/_hl/dataset.py

```diff
@@ -1115,17 +1115,15 @@ def __array__(self, dtype=None, copy=None):
     @with_phil
     def __repr__(self):
         if not self:
-            r = '<Closed HDF5 dataset>'
+            return "<Closed HDF5 dataset>"
+
+        if self.name is None:
+            name = "(anonymous)"
         else:
-            if self.name is None:
-                namestr = '("anonymous")'
-            else:
-                name = pp.basename(pp.normpath(self.name))
-                namestr = '"%s"' % (name if name != '' else '/')
-            r = '<HDF5 dataset %s: shape %s, type "%s">' % (
-                namestr, self.shape, self.dtype.str
-            )
-        return r
+            name = pp.basename(pp.normpath(self.name))
+            name = f'"{name}"'
+
+        return f'<HDF5 dataset {name}: shape {self.shape}, type "{self.dtype.str}">'
```

Comment:
if we are playing code-golf I think this can all be folded into a ternary in the fstring 🤣

Reply:
I om not sure we should be playing code golf though.

Reply:
I think it would become less readable.